aten/src/ATen/native/cuda/BatchLinearAlgebra.cu

```diff
@@ -1625,26 +1625,37 @@ AT_ERROR("cholesky: MAGMA library not found in "
 
 Tensor _cholesky_helper_cuda(const Tensor& self, bool upper) {
   std::vector<int64_t> infos(batchCount(self), 0);
-  Tensor self_working_copy;
-  if (upper) {
-    self_working_copy = cloneBatchedColumnMajor(self.transpose(-1, -2));
+
+  Tensor result;
+  if (self.dim() > 2) {
+    // MAGMA's batched cholesky operator has a off-by-one error
+    // causing IMA (see https://github.com/pytorch/pytorch/issues/42666).
+    // The workaround here is to pad the input with one extra element which
+    // is hidden from the user by resizing the tensor down. This way if MAGMA
+    // reads off bounds it will still be valid user memory.
```

Comment:
Nit: This comment is cool but add an extra line or two about how you're folding the cloneBatchedColumnMajor call into this branch, too